board/BezierBoard.java

```diff
@@ -2413,266 +2379,6 @@ public Object clone()
-				{
-					for(int h = 0; h<1; h++)
-					{
-						System.out.printf("update3DModel() on rails: normals[%d] %f,%f,%f\n", h, normals[h].x, normals[h].y, normals[h].z);
-					}
-				}
-				
-				//Build one quad
-				quads.setCoordinates(nrOfQuads*4, vertices);
-				quads.setNormals(nrOfQuads*4, normals);
-				nrOfQuads++;
-				
-				//Mirror
-				for(int n = 0; n < 4; n++)
-				{
-					mirrorVertices[n].set(vertices[3-n]);
-					mirrorNormals[n].set(normals[3-n]);
-					
-					mirrorVertices[n].y = -mirrorVertices[n].y; 
-					mirrorNormals[n].y = -mirrorNormals[n].y;
-				}
-				
-				//Build mirrored quad
-				quads.setCoordinates(nrOfQuads*4, mirrorVertices);
-				quads.setNormals(nrOfQuads*4, mirrorNormals);
-				nrOfQuads++;
-
-				//Get ready for next step
-				vertices[0].set(vertices[1]);
-				normals[0].set(normals[1]);
-
-				vertices[3].set(vertices[2]);
-				normals[3].set(normals[2]);
-
-				xPos += lengthStep;
-				
-			}
-		}		
-*/
-		minAngle = maxAngle; 
-		maxAngle = 360.0; 
-		for(int i = 0; i < bottomSteps; i++)
-		{	
-			xPos = 0.0;
-			
-			//first coords in lengthwise strip
-			vertices[0].set(getSurfacePoint(xPos, minAngle, maxAngle, i, bottomSteps));
-			normals[0].set(getSurfaceNormal(xPos, minAngle, maxAngle, i, bottomSteps));
-
-			vertices[3].set(getSurfacePoint(xPos, minAngle, maxAngle, i+1, bottomSteps));
-			normals[3].set(getSurfaceNormal(xPos, minAngle, maxAngle, i+1, bottomSteps));
-			
-
-			xPos += lengthStep;
-
-			for(int j = 1; j <= lengthSteps; j++)
-			{	
-				//Two next coords
-				vertices[1].set(getSurfacePoint(xPos, minAngle, maxAngle, i, bottomSteps));
-				normals[1].set(getSurfaceNormal(xPos, minAngle, maxAngle, i, bottomSteps));
-
-				vertices[2].set(getSurfacePoint(xPos, minAngle, maxAngle, i+1, bottomSteps));
-				normals[2].set(getSurfaceNormal(xPos, minAngle, maxAngle, i+1, bottomSteps));
-
-//				if(j == lengthSteps/2)
-//				{
-//					for(int h = 0; h<1; h++)
-//					{
-//						System.out.printf("update3DModel() on bottom: normals[%d] %f,%f,%f\n", h, normals[h].x, normals[h].y, normals[h].z);
-//					}
-//				}
-				
-				//Build one quad
-				quads.setCoordinates(nrOfQuads*4, vertices);
-				quads.setNormals(nrOfQuads*4, normals);
-				nrOfQuads++;
-				
-				//Mirror
-				for(int n = 0; n < 4; n++)
-				{
-					mirrorVertices[n].set(vertices[3-n]);
-					mirrorNormals[n].set(normals[3-n]);
-					
-					mirrorVertices[n].y = -mirrorVertices[n].y; 
-					mirrorNormals[n].y = -mirrorNormals[n].y;
-				}
-				
-				//Build mirrored quad
-				quads.setCoordinates(nrOfQuads*4, mirrorVertices);
-				quads.setNormals(nrOfQuads*4, mirrorNormals);
-				nrOfQuads++;
-
-				//Get ready for next step
-				vertices[0].set(vertices[1]);
-				normals[0].set(normals[1]);
-
-				vertices[3].set(vertices[2]);
-				normals[3].set(normals[2]);
-
-				xPos += lengthStep;
-				
-			}
-		}		
-
-		model.setGeometry(quads);
-
-	}
-	
 	public String toString()
 	{
 		String str = new String();
```

Comment:
Our static analysis engine detects that your pull request includes a change that **method invokes inefficient new String() constructor**. According to [the best practice](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-constructor-dm-string-void-ctor), creating a new java.lang.String object using the no-argument constructor **wastes memory** because the object so created will be functionally indistinguishable from the empty string constant "".  Java guarantees that identical string constants will be represented by the same String object.  **Therefore, you should just use the empty string constant directly.**